app/src/main/java/chat/rocket/android/chatroom/presentation/ChatRoomPresenter.kt

```diff
@@ -468,49 +467,82 @@ class ChatRoomPresenter @Inject constructor(
 
     fun uploadImage(roomId: String, mimeType: String, uri: Uri, bitmap: Bitmap, msg: String) {
         launchUI(strategy) {
-            view.showLoading()
+            val fileName = uriInteractor.getFileName(uri) ?: uri.toString()
             try {
                 withContext(Dispatchers.Default) {
-                    val fileName = uriInteractor.getFileName(uri) ?: uri.toString()
                     if (fileName.isEmpty()) {
                         view.showInvalidFileMessage()
                     } else {
                         val byteArray =
                             bitmap.getByteArray(mimeType, 60, settings.uploadMaxFileSize())
-                        retryIO("uploadFile($roomId, $fileName, $mimeType") {
-                            client.uploadFile(
-                                roomId,
-                                fileName,
-                                mimeType,
-                                msg,
-                                description = fileName
-                            ) {
-                                byteArray.inputStream()
-                            }
-                        }
 
+                        val id = UUID.randomUUID().toString()
+                        val username = userHelper.username()
+                        val user = userHelper.user()
+
+                        val attachment = Attachment(title=fileName, type="file", description = fileName, titleLink=uri.toString(), titleLinkDownload = true, imageUrl = uri.toString())
+                        val message = Message(
+                                id = id,
+                                attachments = listOf(attachment),
+                                roomId=roomId,
+                                timestamp = Instant.now().toEpochMilli(),
+                                sender = SimpleUser(user?.id, user?.username ?: username, user?.name),
+                                avatar = currentServer.avatarUrl(username ?: ""),
+                                synced = false,
+                                unread = true
+                        )
+                        messagesRepository.save(message)
+                        view.showNewMessage(
```

Comment:
This is where the attachment object breaks the change designed to remove the filename from the attachment.